### PR TITLE
Add RecordType.TryGetCapabilities

### PR DESCRIPTION
**Pull request by MikeStall**
Minor fixes to enable creating CDP over RecordType. 
Add method to get DelegationInfo back from a RecordType.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Connectors/Tabular/CdpTableResolver.cs [hunk 1]
```diff
@@ -92,7 +92,8 @@ public async Task<ConnectorType> ResolveTableAsync(string tableName, Cancellatio
             //    sqlRelationships = GetSqlRelationships(text2);
             //}
 
-            string connectorName = _uriPrefix.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries)[1];
+            var parts = _uriPrefix.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Connectors/Tabular/CdpTableResolver.cs:95]: @LucGenetier - fyi

File: src/libraries/Microsoft.PowerFx.Core/Public/Types/RecordType.cs [hunk 2]
```diff
@@ -66,6 +66,25 @@ public RecordType(DisplayNameProvider displayNameProvider, TableDelegationInfo d
 
         private readonly IEnumerable<string> _fieldNames = null;
 
+        /// <summary>
+        /// For tooling, return back capabilities of the table.
+        /// May return null if the table was not created with <see cref="TableDelegationInfo"/>.
+        /// </summary>
+        /// <param name="delegationInfo">The delegation info this table was created with.</param>
+        /// <returns>true if delegationInfo is not null.</returns>
+        public bool TryGetCapabilities(out TableDelegationInfo delegationInfo)
```
[the remark this answers is not in the document]
Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Core/Public/Types/RecordType.cs:75]: Maybe add a comment to say that's only for CDP and that CDS won't work

File: src/tests/Microsoft.PowerFx.Core.Tests.Shared/DelegationTests.cs [hunk 3]
```diff
@@ -40,6 +40,26 @@ public void CapabilityTest()
             Assert.False(neq2);
         }
 
+        [Fact]
+        public void GetCapabilityTest()
+        {
+            RecordType rt = new TestRecordType("myTable", RecordType.Empty().Add("logic", FormulaType.String, "display"), null);
+
+            var ok = rt.TryGetCapabilities(out var delegationInfo);
```
Comment (LucGenetier) [inline src/tests/Microsoft.PowerFx.Core.Tests.Shared/DelegationTests.cs:48]: >delegationInfo [](http://example.com/codeflow?start=51&length=14)

Maybe add a test to check delegationInfo's content

**Review by LucGenetier — APPROVED** [review]
:shipit:

**Review by jas-valgotar — DISMISSED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by LucGenetier — APPROVED** [review]

**Review by jas-valgotar — APPROVED** [review]

**Pull request merged**